src/integration_tests/action_hover_sync.cpp

```diff
@@ -38,14 +38,19 @@ void takeoff_and_hover_at_altitude(float altitude_m)
     ASSERT_TRUE(poll_condition_with_timeout(
         [&dc]() { return dc.is_connected(); }, std::chrono::seconds(10)));
 
-    System& system = dc.system();
-    auto telemetry = std::make_shared<Telemetry>(system);
+    auto systems = dc.systems();
+
+    ASSERT_EQ(systems.size(), 1);
+
+    auto system = systems.at(0);
+
+    auto telemetry = std::make_shared<Telemetry>(*system.lock());
```

Comment:
I'm pretty sure this is wrong because we create a temporary `std::shared_ptr` using `std::weak_ptr::lock()`. The temporary pointer is deleted again, so the reference count goes back to 1 (for MavsdkImpl who owns it).
The plugin is given a reference to the actual `System` and assumes it's valid when that's theoretically not true.